internal/storage/webhook/webhook.go

```diff
@@ -0,0 +1,169 @@
+package webhookstorage
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/darchlabs/synchronizer-v2/internal/storage"
+	"github.com/darchlabs/synchronizer-v2/pkg/webhook"
+	"github.com/pkg/errors"
+)
+
+type Storage struct {
+	storage *storage.S
+}
+
+func New(s *storage.S) *Storage {
+	return &Storage{
+		storage: s,
+	}
+}
+
+func (s *Storage) CreateWebhook(wh *webhook.Webhook) (*webhook.Webhook, error) {
+	query := `
+		INSERT INTO webhooks (id, entity_type, entity_id, endpoint, payload, created_at, updated_at, sent_at, next_retry_at) 
+		VALUES (:id, :entity_type, :entity_id, :endpoint, :payload, :created_at, :updated_at, :sent_at, :next_retry_at)
+		RETURNING id
+	`
+
+	rows, err := s.storage.DB.NamedQuery(query, wh)
+	if err != nil {
+		return nil, errors.Wrap(err, "webhookstorage: error creating webhook")
+	}
+	defer rows.Close()
+
+	if rows.Next() {
+		var id string
+		if err := rows.Scan(&id); err != nil {
+			return nil, errors.Wrap(err, "webhookstorage: error scanning webhook ID")
+		}
+		wh.ID = id
+	} else {
+		return nil, errors.New("webhookstorage: no ID returned after webhook creation")
+	}
+
+	createdWebhook, err := s.GetWebhookByID(wh.ID)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
is there 100% necessary to use this kind of logic? Why are we using the returning statement if we already have the ID ? Another thing to notice here is that ir super and pretty uncommon thing to receive a data type and return the exact same data type. Something pretty common for this insert methods is to receive a "record" to insert and only return an error since the input record should have the entire data needed for this